tests/integration/test_integration.py

```diff
@@ -418,3 +418,37 @@ def test_flush_log(caplog):
             )
         ]
         log.log.assert_has_calls(calls)
+
+
+def test_regression_logging_in_context(tmpdir):
```

Comment:
This regression test should deadlock

Reply:
Confirmed here: https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/8439/workflows/8b987cb4-8a37-44d4-af2a-7166bc66ee10/jobs/693155